src/Injector.php

```diff
@@ -25,18 +25,27 @@
 final class Injector
 {
     private ContainerInterface $container;
-    private bool $cacheReflections;
+    private bool $cacheReflections = false;
 
     /**
      * @var ReflectionClass[]
      * @psalm-var array<class-string,ReflectionClass>
      */
     private array $reflectionsCache = [];
 
-    public function __construct(ContainerInterface $container, bool $cacheReflections = false)
+    public function __construct(ContainerInterface $container)
     {
         $this->container = $container;
-        $this->cacheReflections = $cacheReflections;
+    }
+
+    /**
+     * Caches reflection objects in memory for performance.
+     */
+    public function withCacheReflections(bool $cacheReflections): self
```

Comment:
I think it will be better name: `withReflectionCache`

Reply:
I'd agree if `withReflectionCache` was a setter for a cache instance

Reply:
> I'd agree if `withReflectionCache` was a setter for a cache instance

I'd agree if it would be `withReflectionCacher`